src/chainstate/burn/db/sortdb.rs

```diff
@@ -2254,13 +2397,85 @@ impl <'a> SortitionHandleTx <'a> {
                     keys.push(db_keys::pox_last_anchor().to_string());
                     values.push("".to_string());
                 }
+                // if we've selected an anchor _and_ know of the anchor,
+                //  write the reward set information
+                if let Some(mut reward_set) = reward_info.known_selected_anchor_block_owned() {
+                    if reward_set.len() > 0 {
+                        // if we have a reward set, then we must also have produced a recipient
+                        //   info for this block
+                        let mut recipients_to_remove: Vec<_> = recipient_info.unwrap().recipients.iter().map(
```

Comment:
If we're going to `unwrap()`, this should `reward_info` and `recipient_info` be combined into a single `Option<..>`?

Reply:
`recipient_info` is only guaranteed to be `Some` if reward_info is `Some` _and_ `known_selected_anchor_block_owned` is `Some`, so tying them into the same object would require adding a new enum type for shuttling info specifically to this function.